jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/FileOperations.java

```diff
@@ -52,5 +57,27 @@ public static void copy(ImmutableList<Path> sourceFiles, Path destDir) throws IO
     }
   }
 
+  /**
+   * Acquires an exclusive lock on the {@code file} and opens an {@link OutputStream} to write to
```

Comment:
I think should make our intentions a little more clear here on why we're locking. They are an assertion rather than concurrency management feature?